openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,529 @@
+                        'stage': -1
+                    },
+                    status=400
+                )
+            finally:
+                tar_file.close()
+
+            log.info(
+                "Course import {0}: Uploaded file extracted".format(courselike_key)
+            )
+            self._save_request_status(request, courselike_string, 2)
+
+            # find the 'course.xml' file
+            def get_all_files(directory):
+                """
+                For each file in the directory, yield a 2-tuple of (file-name,
+                directory-path)
+                """
+                for dirpath, _dirnames, filenames in os.walk(directory):
+                    for filename in filenames:
+                        yield (filename, dirpath)
+
+            def get_dir_for_fname(directory, filename):
+                """
+                Returns the dirpath for the first file found in the directory
+                with the given name.  If there is no file in the directory with
+                the specified name, return None.
+                """
+                for fname, dirpath in get_all_files(directory):
+                    if fname == filename:
+                        return dirpath
+                return None
+
+            dirpath = get_dir_for_fname(course_dir, root_name)
+            if not dirpath:
+                self._save_request_status(request, courselike_string, -2)
+                return JsonResponse(
+                    {
+
+                        'error_message': _(
+                            'Could not find the {0} file in the package.'.
+                                format(root_name)
+                        ),
+                        'stage': -2
+                    },
+                    status=415
+                )
+
+            dirpath = os.path.relpath(dirpath, data_root)
+            logging.debug('found %s at %s', root_name, dirpath)
+
+            log.info(
+                "Course import %s: Extracted file verified",
+                courselike_key
+            )
+            self._save_request_status(request, courselike_string, 3)
+
+            courselike_items = import_func(
+                modulestore(),
+                request.user.id,
+                settings.GITHUB_REPO_ROOT,
+                [dirpath],
+                load_error_modules=False,
+                static_content_store=contentstore(),
+                target_id=courselike_key,
+            )
+
+            new_location = courselike_items[0].location
+            logging.debug('new course at {0}'.format(new_location))
+
+            log.info(
+                "Course import {0}: Course import successful".format(courselike_key)
+            )
+            self._save_request_status(request, courselike_string, 4)
+
+        # Send errors to client with stage at which error occurred.
+        except Exception as exception:   # pylint: disable=W0703
```

Comment:
pylint warnings are each identified by a alphanumeric code (C0112) and a symbolic name (empty-docstring). Prefer the symbolic names in new code or when updating existing code.